tests/vmi_lifecycle_test.go

```diff
@@ -435,6 +437,39 @@ var _ = Describe("[rfe_id:273][crit:high][arm64][vendor:cnv-qe@redhat.com][level
 			})
 		})
 
+		Context("with nodeselector", func() {
```

Comment:
I don't think that adding these tests does make too much sense.

virt-handler is the one which marks nodes as a potential target vor VMIs. It does so, by setting this label on nodes: `kubevirt.io/schedulable: true`. If it detects incompatibilities it will set it to false. If virt-handler itself crashes, virt-controller will detect the missing heartbeat of virt-handler and set it to false on the corresponding nodes too.

This ensures that we only schedule on "properly working" nodes. Now: virt-handler and our virt-stack are bound to linux and can therefore ever only report linux nodes as ready. That `linux` is one of the key factors (others are kvm compatible hardware, certain linux modules, ...). That are however only "implementation details". What matters is the `kubevirt.io/schedulable: true` label. 

If people choose to add a `windows` node selector to their scheduling requirements, it will just not be scheduled due to virt-handler not being there and not reporting the node as "kubevirt compatible". That is like with any other node selector and missing labels on that node.

I personally would not add these tests. It is not really clear to me what desired behaviour this is preserving.

Reply:
These tests should should check if kubevirt will not modify user defined node selector. In first test we check negative result. And thats why I defined `k8sv1.LabelOSStable: "not-existing-os"` to be sure, kubevirt will not change it and vm will be in pending state, because there might be in future functionality, which forces some default node selector. Second test check if vm is scheduled and running with correct node selector, then the test checks if virt-launcher pod has correct node selector.

Reply:
Ok, I could not believe that we are not explicitly testing node selectors and therefore thought this was really about testing something in context of the node OS. Since I could not find any explicit tests (only implicit ones and explicit tests for affinity), makes sense to me. Revoking my comment. :+1: